content/en/blog/_posts/2025-02-03-introducing-jobset/2025-02-03-introducing-jobset.md

```diff
@@ -0,0 +1,211 @@
+not adequately model them alone anymore. 
+
+Furthermore, the landscape of Kubernetes distributed training orchestration APIs has become
+fragmented, and each of the existing solutions in this fragmented landscape has certain limitations
+that make it non-optimal for distributed ML training. 
+
+For example, the KubeFlow training operator defines custom APIs for different frameworks (e.g.
+PyTorchJob, TFJob, MPIJob, etc.); however, each of these job types are in fact a solution fit
+specifically to the target framework, each with different semantics and behavior. 
+
+On the other hand, the Job API fixed many gaps for running batch workloads, including Indexed
+completion mode, higher scalability, Pod failure policies and Pod backoff policy to mention a few of
+the most recent enhancements. However, running ML training and HPC workloads using the upstream Job
+API requires extra orchestration to fill the following gaps:
+
+Multi-template Pods : Most HPC or ML training jobs include more than one type of Pods. The different
+Pods are part of the same workload, but they need to run a different container, request different
+resources or have different failure policies. A common example is the driver-worker pattern.
+
+Job groups : Large scale training workloads span multiple network topologies, running across
+multiple racks for example. Such workloads are network latency sensitive, and aim to localize
+communication and minimize traffic crossing the higher-latency network links. To facilitate this,
+the workload needs to be split into groups of Pods each assigned to a network topology.
+
+Inter-Pod communication : Create and manage the resources (e.g. [headless
+Services](/docs/concepts/services-networking/service/#headless-services)) necessary to establish
+communication between the Pods of a job.
+
+Startup sequencing : Some jobs require a specific start sequence of pods; sometimes the driver is
+expected to start first (like Ray or Spark), in other cases the workers are expected to be ready
+before starting the driver (like MPI).
+
+JobSet aims to address those gaps using the Job API as a building block to build a richer API for
+large-scale distributed HPC and ML use cases.
+
+## How JobSet Works
+JobSet models a distributed batch workload as a group of Kubernetes Jobs. This allows a user to
+easily specify different pod templates for different distinct groups of pods (e.g. a leader,
+workers, parameter servers, etc.). 
+
+It uses the abstraction of a ReplicatedJob to manage child Jobs, where a ReplicatedJob is
+essentially a Job Template with some desired number of Job replicas specified. This provides a
+declarative way to easily create identical child-jobs to run on different islands of accelerators,
+without resorting to scripting or Helm charts to generate many versions of the same job but with
+different names.
+
+<figure>
+  <img src="jobset_diagram.svg">
```

Comment:
This doesn't render when I preview the article. I'm afraid that working images are a prerequisite for a merge.

Please consider using a [figure](https://gohugo.io/content-management/shortcodes/#figure) short code with an `alt` attribute set.

Reply:
Would replacing this block with the following work?:

```
{{< figure src="jobset_diagram.svg" alt="JobSet Architecture" class="diagram-large" clicktozoom="true" >}}
```

or

```
![JobSet Architecture](jobset_diagram.svg)
```

Reply:
> Would replacing this block with the following work?
> 
> ```
> {{< figure src="jobset_diagram.svg" alt="JobSet Architecture" class="diagram-large" clicktozoom="true" >}}
> ```
> 
> or
> 
> ```
> ![JobSet Architecture](jobset_diagram.svg)
> ```

Yes I previously used the 2nd suggestion there but the review feedback was to use figure shortcode, which I guess I used incorrectly here. Trying the 1st suggestion now.

Reply:
@ahg-g @sftim let me know if this rendering looks good now. When I view the netlify preview I actually can't find the blog post. I tried entering what I think is the URL manually (https://deploy-preview-45759--kubernetes-io-main-staging.netlify.app/blog/2025/02/03/introducing-jobset/) as well as searching through the blog posts, but could't find it.

Reply:
I tested it and it didn't work, it turns out you need to name the file to `index.md`, if you do that it works

![image](https://github.com/user-attachments/assets/3c12dba7-406a-43a3-a0e2-f93cff3029b5)

Reply:
Thanks, renamed to index.md